### PR TITLE
feat(frontend): support \d, \dt and \dm

### LINKED ISSUE #5128: feat: support \d, \dt, \dm
### Is your feature request related to a problem? Please describe.

#2954

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As title, support `\d`、`\dt` and `\dm` in psql:
```sql
SELECT n.nspname as "Schema",
  c.relname as "Name",
  CASE c.relkind WHEN 'r' THEN 'table' WHEN 'v' THEN 'view' WHEN 'm' THEN 'materialized view' WHEN 'i' THEN 'index' WHEN 'S' THEN 'sequence' WHEN 's' THEN 'special' WHEN 't' THEN 'TOAST table' WHEN 'f' THEN 'foreign table' WHEN 'p' THEN 'partitioned table' WHEN 'I' THEN 'partitioned index' END as "Type",
  pg_catalog.pg_get_userbyid(c.relowner) as "Owner"
FROM pg_catalog.pg_class c
     LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
WHERE c.relkind IN ('r','p','v','m','S','f','')
      AND n.nspname <> 'pg_catalog'
      AND n.nspname !~ '^pg_toast'
      AND n.nspname <> 'information_schema'
  AND pg_catalog.pg_table_is_visible(c.oid)
ORDER BY 1,2;

SELECT n.nspname as "Schema",
  c.relname as "Name",
  CASE c.relkind WHEN 'r' THEN 'table' WHEN 'v' THEN 'view' WHEN 'm' THEN 'materialized view' WHEN 'i' THEN 'index' WHEN 'S' THEN 'sequence' WHEN 's' THEN 'special' WHEN 't' THEN 'TOAST table' WHEN 'f' THEN 'foreign table' WHEN 'p' THEN 'partitioned table' WHEN 'I' THEN 'partitioned index' END as "Type",
  pg_catalog.pg_get_userbyid(c.relowner) as "Owner"
FROM pg_catalog.pg_class c
     LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
WHERE c.relkind IN ('r','p','')
      AND n.nspname <> 'pg_catalog'
      AND n.nspname !~ '^pg_toast'
      AND n.nspname <> 'information_schema'
  AND pg_catalog.pg_table_is_visible(c.oid)
ORDER BY 1,2;

SELECT n.nspname as "Schema",
  c.relname as "Name",
  CASE c.relkind WHEN 'r' THEN 'table' WHEN 'v' THEN 'view' WHEN 'm' THEN 'materialized view' WHEN 'i' THEN 'index' WHEN 'S' THEN 'sequence' WHEN 's' THEN 'special' WHEN 't' THEN 'TOAST table' WHEN 'f' THEN 'foreign table' WHEN 'p' THEN 'partitioned table' WHEN 'I' THEN 'partitioned index' END as "Type",
  pg_catalog.pg_get_userbyid(c.relowner) as "Owner"
FROM pg_catalog.pg_class c
     LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
WHERE c.relkind IN ('m','')
      AND n.nspname <> 'pg_catalog'
      AND n.nspname !~ '^pg_toast'
      AND n.nspname <> 'information_schema'
  AND pg_catalog.pg_table_is_visible(c.oid)
ORDER BY 1,2;
```
image

In this PR:
1. implement `pg_catalog.pg_get_userbyid` and rewrite it as subquery.
2. Temporarily implement `pg_table_is_visible` by simply return `true`, since `pg_class` only returns all relations under connected database.
3. fix `relkind` format in `pg_class` and make it compatible with PG, note that, `(materialized) source` are not supported yet in `\d`: 
     `r = ordinary table, i = index, S = sequence, t = TOAST table, v = view, m = materialized view, c = composite type, f = foreign table, p = partitioned table, I = partitioned index`
4. add binary operation support for `PGRegexMatch` and `PGRegexNotMatch`.

Besides, for different version of `psql`, the sql generated by `\d` might be a little bit different.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* SQL commands, functions, and operators
support some `psql` commands:
1. `\d`: list all relations in current database, `(materialized) source` is not supported yet, together with indexes(originally not supported in psql).
2. `\dt`: list all tables in current database.
3. `\dm`: list all materialized views in current database.

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
Resolve #5128 
Resolve partially #2954

### CODE REVIEW

**Review by TennyZhuang — APPROVED** [review]

File: src/frontend/src/binder/expr/binary_op.rs [hunk 1]
```diff
@@ -89,4 +93,20 @@ impl Binder {
         };
         Ok(FunctionCall::new(func_type, vec![left, right])?.into())
     }
+
+    fn bind_regex_match(&mut self, left: ExprImpl, right: ExprImpl) -> Result<ExprImpl> {
+        Ok(FunctionCall::new(
+            ExprType::IsNotNull,
```
Comment (xiangjinwu) [inline src/frontend/src/binder/expr/binary_op.rs:99]: This rewrite is not equivalent when either `left` or `right` is `null`. The operator returns `null` but this returns `false`. Let's open an issue to track the fix.

File: src/frontend/src/binder/select.rs [hunk 2]
```diff
@@ -15,16 +15,22 @@
 use std::fmt::Debug;
 
 use itertools::Itertools;
-use risingwave_common::catalog::{Field, Schema};
+use risingwave_common::catalog::{Field, Schema, PG_CATALOG_SCHEMA_NAME};
 use risingwave_common::error::{ErrorCode, Result};
 use risingwave_common::types::DataType;
+use risingwave_pb::expr::expr_node::Type;
```
Comment (xiangjinwu) [inline src/frontend/src/binder/select.rs:21]: `crate::expr::ExprType`

File: src/frontend/src/binder/select.rs [hunk 3]
```diff
@@ -238,6 +244,55 @@ impl Binder {
         Ok((select_list, aliases))
     }
 
+    /// `bind_get_user_by_id_select` binds a select statement that returns a single user name by id,
+    /// this is used for function `pg_catalog.get_user_by_id()`.
+    pub fn bind_get_user_by_id_select(&mut self, input: &ExprImpl) -> Result<BoundSelect> {
+        let select_items = vec![ExprImpl::InputRef(Box::new(InputRef::new(
```
Comment (xiangjinwu) [inline src/frontend/src/binder/select.rs:250]: FYI `ExprImpl::InputRef(Box::new(e))` can be replaced with `e.into()` for all expr variants.

File: src/frontend/src/binder/select.rs [hunk 4]
```diff
@@ -238,6 +244,55 @@ impl Binder {
...
+        )))];
+        let schema = Schema {
+            fields: vec![Field::with_name(
+                DataType::Varchar,
+                UNNAMED_COLUMN.to_string(),
+            )],
+        };
+        let input = match input {
+            ExprImpl::InputRef(input_ref) => ExprImpl::CorrelatedInputRef(Box::new(
+                CorrelatedInputRef::new(input_ref.index, input_ref.data_type.clone(), 1),
```
Comment (xiangjinwu) [inline src/frontend/src/binder/select.rs:262]: keep consistent with the arm below (`.index()` and `.return_type()`)

File: src/frontend/src/binder/select.rs [hunk 5]
```diff
@@ -238,6 +244,55 @@ impl Binder {
...
+            )),
+            ExprImpl::CorrelatedInputRef(col_input_ref) => {
+                ExprImpl::CorrelatedInputRef(Box::new(CorrelatedInputRef::new(
+                    col_input_ref.index(),
+                    col_input_ref.return_type(),
+                    col_input_ref.depth() + 1,
+                )))
+            }
+            ExprImpl::Literal(_) => input.clone(),
+            _ => unimplemented!("Unsupported input type"),
```
Comment (xiangjinwu) [inline src/frontend/src/binder/select.rs:272]: `return Err` rather than panic. The message can also refer to `get_user_by_id` to ease debugging.

**Comment by fuyufjh** [comment]
Please fill the "Documentation" section in the PR description ❤️

**Comment by fuyufjh** [comment]
BTW, `\d <table_name>` is not supported.

**Comment by yezizp2012** [comment]
> BTW, `\d <table_name>` is not supported.

NTFS, we still need to support `OPERATOR` like postgreSQL for that. The actual query is as follows:
```sql
SELECT c.oid,
  n.nspname,
  c.relname
FROM pg_catalog.pg_class c
     LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
WHERE c.relname OPERATOR(pg_catalog.~) '^(t)$'
  AND pg_catalog.pg_table_is_visible(c.oid)
ORDER BY 2, 3;
```

**Pull request merged**